### PR TITLE
HBASE-23171 Ensure region state error messages include server making report (branch-2.1)

### PR DESCRIPTION
**Pull request by busbey**

### CODE REVIEW

**Comment by binlijin** [comment]
+1

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-2.1 Compile Tests _ |
| :green_heart: |  mvninstall  |   5m 31s |  branch-2.1 passed  |
| :green_heart: |  compile  |   0m 55s |  branch-2.1 passed  |
| :green_heart: |  checkstyle  |   1m 31s |  branch-2.1 passed  |
| :green_heart: |  shadedjars  |   4m 18s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   0m 34s |  branch-2.1 passed  |
| :blue_heart: |  spotbugs  |   2m 56s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   2m 55s |  branch-2.1 passed  |
||| _ Patch Compile Tests _ |
| :green_heart: |  mvninstall  |   5m  7s |  the patch passed  |
| :green_heart: |  compile  |   0m 57s |  the patch passed  |
| :green_heart: |  javac  |   0m 57s |  the patch passed  |
| :green_heart: |  checkstyle  |   1m 26s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   4m 21s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |  21m 45s |  Patch does not cause any errors with Hadoop 2.7.7 2.8.5 or 3.0.3 3.1.2.  |
| :green_heart: |  javadoc  |   0m 33s |  the patch passed  |
| :green_heart: |  findbugs  |   3m  7s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  | 200m 32s |  hbase-server in the patch passed.  |
| :green_heart: |  asflicense  |   0m 25s |  The patch does not generate ASF License warnings.  |
|  |   | 261m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.4 Server=19.03.4 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-777/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/777 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 51af2e7626df 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-777/out/precommit/personality/provided.sh |
| git revision | branch-2.1 / 3f12d50282 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-777/1/testReport/ |
| Max. process+thread count | 4692 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-777/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java [hunk 1]
```diff
@@ -991,8 +991,8 @@ void checkOnlineRegionsReportForMeta(final ServerName serverName, final Set<byte
         final RegionInfo hri = getMetaRegionFromName(regionName);
         if (hri == null) {
           if (LOG.isTraceEnabled()) {
-            LOG.trace("Skip online report for region=" + Bytes.toStringBinary(regionName) +
-              " while meta is loading");
+            LOG.trace("Skip online report for region={} while meta is loading from server={}",
```
Comment (saintstack) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java]: region= is redundant.

Comment (saintstack) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java]: So is server= ... smile. Who else would be making a report here.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java [hunk 2]
```diff
@@ -1035,7 +1035,8 @@ void checkOnlineRegionsReport(final ServerStateNode serverNode, final Set<byte[]
                     " but state has otherwise AND NO procedure is running");
                 }
               } catch (UnexpectedStateException e) {
-                LOG.warn(regionNode.toString() + " reported unexpteced OPEN: " + e.getMessage(), e);
+                LOG.warn("{} reported unexpteced OPEN: {} sever={}", regionNode.toString(),
```
Comment (saintstack) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java]: spelling

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java (hunk 1 above)
Comment (busbey) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java]: Why would the server making the report be redundant?

Comment (busbey) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/AssignmentManager.java]: If regionName is redundant then why does `getMetaRegionFromName`  take it as an argument?

**Pull request merged**